airflow/providers/jenkins/operators/jenkins_job_trigger.py

```diff
@@ -195,9 +201,10 @@ def execute(self, context):
             'Triggering the job %s on the jenkins : %s with the parameters : %s',
             self.job_name, self.jenkins_connection_id, self.parameters)
         jenkins_server = self.get_hook().get_jenkins_server()
-        jenkins_response = self.build_job(jenkins_server)
-        build_number = self.poll_job_in_queue(
-            jenkins_response['headers']['Location'], jenkins_server)
+        jenkins_response = self.build_job(jenkins_server, self.parameters)
```

Comment:
Here's where we preserve the same behavior for `self.build_job` by passing `self.parameters` as the "local" parameters in the function.